sphinx/source/start_network.rst

```diff
@@ -1,135 +1,102 @@
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The :ref:`governance` rules are defined as a Lua script passed via the ``--gov-script`` option. For example, a default set of `governance rules <https://github.com/microsoft/CCF/blob/master/src/runtime_config/gov.lua>`_ can be used to define a majority of members as the quorum of the consortium.
 
-Once the initial set of nodes is running, the ``nodes.json`` file specifying the configuration of the original network should be created. For example, for a network of two nodes, the ``nodes.json`` file will be:
+The identities of members and users are specified as `glob patterns <https://en.wikipedia.org/wiki/Glob_(programming)>`_ via the ``--member-certs`` and ``--user-certs`` option, respectively. For example, if 2 members (``member1_cert.pem`` and ``member2_cert.pem``) and 3 users (``user1_cert.pem``, ``user2_cert.pem`` and ``user3_cert.pem``) should be added to CCF, operators should specify ``--member-certs=member*_cert.pem`` and ``--user-certs=user*_cert.pem``.
 
-.. code-block:: bash
-
-    $ cat nodes.json
-    [
-        {
-            "pubhost": "rpc_public_ip0",
-            "cert": [<output node0 cert bytes>],
-            "host": "node/rpc_ip0",
-            "quote": [<output quote0 bytes>],
-            "status": 0,
-            "nodeport": "node_port0",
-            "rpcport": "rpc_port0"
-        },
-        {
-            "pubhost": "rpc_public_ip1",
-            "cert": [<output node1 cert bytes>],
-            "host": "node/rpc_ip1",
-            "quote": [<output quote1 bytes>],
-            "status": 0,
-            "nodeport": "node_port1",
-            "rpcport": "rpc_port1"
-        }
-    ]
-
-Then, certificates for members and users can be created to allow secure TLS communication between the clients and the enclaves of each node. For example, for two members and one user, you should run:
-
-.. code-block:: bash
-
-    $ genesisgenerator cert --name=member1
-    $ genesisgenerator cert --name=member2
-    $ genesisgenerator cert --name=user1
-
-Finally, the genesis transaction (``tx0``), containing the initial state of the network, including the initial set of nodes, users and members certificates and governance scripts, can be created:
+.. note:: Once a CCF network is started, members can add other members and users via governance. See :ref:`Submitting a new proposal` for more information.
 
-.. code-block:: bash
-
-    $ genesisgenerator tx --members=member*cert.pem --users=user*cert.pem --nodes=nodes.json --gov-script=src/runtime_config/gov.lua --tx0=tx0 --start-json=startNetwork.json
-
-This command also generates the ``startNetwork.json`` RPC file required to start up the network.
+When CCF is used to run a custom Lua application, the starting node should also be started with the ``--app-script=/path/to/lua/application_script`` (see the `samples folder <https://github.com/microsoft/CCF/tree/master/samples/apps>`_ for example of Lua applications).
```

Comment:
To discuss: This seems odd. We shouldn't really privilege a specific app. Either we should support a generic json file detailing initial modifications to the KV (but that sounds too much like a genesis transaction), or the state should be bootstrappable by members on the running network.

Reply:
Good spot. This will change as part of https://github.com/microsoft/CCF/issues/293 and the documentation updated once more I'm afraid.

Reply:
Setting up the lua app should be something that happens on the network open vote, when using the luagenericapp.